tidy3d/components/geometry.py

```diff
@@ -2015,14 +2017,18 @@ def length_axis(self) -> float:
         zmin, zmax = self.slab_bounds
         return zmax - zmin
 
+    @pydantic.validator("vertices", pre=True, always=True)
+    def convert_to_numpy(cls, val):
```

Comment:
Why do we need this pre-conversion? Using later `copy=true` in the `ArrayLike` validator will most likely lead to 2 copies.

Reply:
it's unfortunate but the reasoning is because a lot of other validators for "vertices" require the vertices in np.ndarray form.

Previously, they were each converting the `val` to `np.ndarray`, but I decided to just refactor in a `pre` validator so that it only occurs once. Of course, there will be an extra conversion from the `ArrayLike` validator. I'm not sure the best workaround here.

Reply:
I see… in that case I think we live with it, then.